base/strings/basic.jl

```diff
@@ -630,6 +653,36 @@ end
     end
     return  _isascii(cu,last-chunk_size+1,last)
 end
+
+function isascii(bytes::AbstractVector{UInt8}, first, last)
+    q_chunk_size = 128
+
+    n = last - first + 1
+    nqwords, epilog_bytes = (n >> 3, n & 0x07) # divrem(n,8)
+    epilog_first = last - epilog_bytes + 1
+    if epilog_bytes > 0
+        _isascii_subqword(bytes,epilog_first,epilog_bytes) || return false
+    end
+    nqwords > 0 || return true
+    nchunks, prolog_qwords = divrem(nqwords,q_chunk_size)
+
+    qwords = @inline @inbounds reinterpret(UInt64,@inbounds view(bytes,first:(epilog_first - 1)))
```

Comment:
```suggestion
    qwords = @inline reinterpret(UInt64, @inbounds view(bytes, first:(epilog_first - 1)))
```

Not 100% sure whether that changes the performance characteristics, but as far as I can tell there's not really anything in `reinterpret` that would need an `@inbounds`.